File: Python3/_94.py
# Left, Node, Right

class Solution:
    def inorderTraversal(self, current: Optional[TreeNode]):
        l, stack = [], []
        
        while current or stack:
            if current:
                stack.append(current)
                current = current.left
                
            else:
                current = stack.pop()
                l.append(current.val)
                current = current.right
                
        return l

File: Cpp/_94.cpp
// Left, Node, Center

class Solution {
public:
    vector<int> inorderTraversal(TreeNode* root) {
        stack<TreeNode*> stack;
        vector<int> l;
        TreeNode* current = root;

        while (current || !stack.empty()) {
            if (current) {
                stack.push(current);
                current = current->left;

            }
            else {
                current = stack.top();
                stack.pop();
                l.push_back(current->val);
                current = current->right;
            }
        }
        return l;
    }
};